### ISSUE TITLE
Connection configuration always chooses configuration with full name

### ISSUE THREAD

**Issue by mikary** [aspnet/EntityFramework#1395]
When using multiple configuration files, if one configuration file specifies a connection string using the full name of the DbContext and the other uses the short name, the configuration from the file with the full name is always selected regardless of the order of the configuration files.

```cs
namespace FullName
{
    public class MyContext : DbContext
    {
        public MyContext(IServiceProvider serviceProvider)
            : base(serviceProvider)
        {
        }
    }

    public class Program
    {
        public void Main(string[] args)
        {
            var configuration = new Configuration();
            configuration.AddIniFile("config1.ini");
            configuration.AddIniFile("config2.ini");

            var serviceProvider = new ServiceCollection()
                .AddEntityFramework(configuration)
                .AddSqlServer()
                .AddDbContext<MyContext>()
                .ServiceCollection
                .BuildServiceProvider();

            using (var context = new MyContext(serviceProvider))
            {
                context.Database.EnsureCreated();
                Console.WriteLine(context.Database.AsRelational().Connection.ConnectionString);
                Console.ReadLine();
            }
        }
    }
}
```

config1.ini:
```ini
[EntityFramework]
FullName.MyContext:ConnectionString = Data Source=(localdb)\v11.0;Initial Catalog=MyDBFull;Integrated Security=True;MultipleActiveResultSets=True;Connect Timeout=30
```

config2.ini
```ini
[EntityFramework]
MyContext:ConnectionString = Data Source=(localdb)\v11.0;Initial Catalog=MyDBShort;Integrated Security=True;MultipleActiveResultSets=True;Connect Timeout=30
```

**Comment by smitpatel** [comment]
We search for keys with short name before full name. Since keys are being overwritten the last one wins so key with full name used.

**Comment by rowanmiller** [comment]
We should discuss with team, there with issues with having one take precedence (which one wins also seems somewhat arbitrary) and there are issues with throwing. Either option makes it impossible to override one with the other.

**Comment by rowanmiller** [comment]
This seems ok, fully qualified taking precedence is fine. Agreed that ordering is a little confusing... but we're ok with it. @mikary can you make sure we have a test that ensures we don't change this behavior.

**Comment by rowanmiller** [comment]
Assigning to @mikary to add a test to ensure we don't change this behavior in the future.

**Comment by smitpatel** [comment]
@rowanmiller - I have already included test for it in pull req #1409

**Comment by rowanmiller** [comment]
Great, feel free to close once pushed

**Comment by smitpatel** [comment]
fixed via 4d6c82826674edfee69ee98c6ee1970ee5e89ee5
